In [1]:
%load_ext cudf.pandas

In [2]:
import sys
import os

sys.path.insert(0, os.path.abspath("/home/ElasticNotebook"))
%load_ext ElasticNotebook
import cudf

Enabled rmm statistics


In [3]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load
# %load_ext cudf.pandas
import os

if "IREWR_WITH_MODIN" in os.environ and os.environ["IREWR_WITH_MODIN"] == "True":
    # STEFANOS: Import Modin Pandas
    import os

    os.environ["MODIN_ENGINE"] = "ray"
    import ray

    ray.init(
        num_cpus=int(os.environ["MODIN_CPUS"]),
        runtime_env={"env_vars": {"__MODIN_AUTOIMPORT_PANDAS__": "1"}},
    )
    import modin.pandas as pd
else:
    # STEFANOS: Import regular Pandas
    import pandas as pd
# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All"
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

In [6]:
%%time
### cell 0 ###

data = pd.read_csv(
    "/home/dias-benchmarks/notebooks/josecode1/billionaires-statistics-2023/input/test.csv",
    index_col="rank",
)

CPU times: user 27.4 ms, sys: 13.6 ms, total: 40.9 ms
Wall time: 27.7 ms


In [7]:
data.info()

<class 'cudf.core.dataframe.DataFrame'>
Index: 2640 entries, 1 to 2540
Data columns (total 34 columns):
 #   Column                                      Non-Null Count  Dtype
---  ------                                      --------------  -----
 0   finalWorth                                  2640 non-null   int64
 1   category                                    2640 non-null   object
 2   personName                                  2640 non-null   object
 3   age                                         2575 non-null   float64
 4   country                                     2602 non-null   object
 5   city                                        2568 non-null   object
 6   source                                      2640 non-null   object
 7   industries                                  2640 non-null   object
 8   countryOfCitizenship                        2640 non-null   object
 9   organization                                325 non-null    object
 10  selfMade                     

In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["finalWorth"] = data["finalWorth"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["category"] = data["category"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["personName"] = data["personName"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["age"] = data["age"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["country"] = data["country"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["city"] = data["city"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["source"] = data["source"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["industries"] = data["industries"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["countryOfCitizenship"] = data["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["organization"] = data["organization"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["selfMade"] = data["selfMade"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["status"] = data["status"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["gender"] = data["gender"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["birthDate"] = data["birthDate"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["lastName"] = data["lastName"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["firstName"] = data["firstName"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["title"] = data["title"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["date"] = data["date"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["state"] = data["state"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["residenceStateRegion"] = data["residenceStateRegion"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["birthYear"] = data["birthYear"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["birthMonth"] = data["birthMonth"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["birthDay"] = data["birthDay"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["cpi_country"] = data["cpi_country"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["cpi_change_country"] = data["cpi_change_country"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["gdp_country"] = data["gdp_country"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["gross_tertiary_education_enrollment"] = data[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["gross_primary_education_enrollment_country"] = data[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["life_expectancy_country"] = data["life_expectancy_country"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["tax_revenue_country_country"] = data["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["total_tax_rate_country"] = data["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["population_country"] = data["population_country"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["latitude_country"] = data["latitude_country"]

In [ ]:
%%time
## Transfer_post 0 ##
pd_data["longitude_country"] = data["longitude_country"]

In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["finalWorth"] = data["finalWorth"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["category"] = data["category"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["personName"] = data["personName"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["age"] = data["age"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["country"] = data["country"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["city"] = data["city"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["source"] = data["source"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["industries"] = data["industries"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["countryOfCitizenship"] = data["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["organization"] = data["organization"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["selfMade"] = data["selfMade"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["status"] = data["status"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["gender"] = data["gender"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["birthDate"] = data["birthDate"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["lastName"] = data["lastName"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["firstName"] = data["firstName"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["title"] = data["title"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["date"] = data["date"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["state"] = data["state"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["residenceStateRegion"] = data["residenceStateRegion"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["birthYear"] = data["birthYear"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["birthMonth"] = data["birthMonth"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["birthDay"] = data["birthDay"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["cpi_country"] = data["cpi_country"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["cpi_change_country"] = data["cpi_change_country"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["gdp_country"] = data["gdp_country"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["gross_tertiary_education_enrollment"] = data[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["gross_primary_education_enrollment_country"] = data[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["life_expectancy_country"] = data["life_expectancy_country"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["tax_revenue_country_country"] = data["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["total_tax_rate_country"] = data["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["population_country"] = data["population_country"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["latitude_country"] = data["latitude_country"]

In [ ]:
%%time
## Transfer_pre 1 ##
pd_data["longitude_country"] = data["longitude_country"]

In [5]:
%%time
### cell 1 ###

factor = 800
data = pd.concat([data] * factor, ignore_index=True)
data.info()

<class 'cudf.core.dataframe.DataFrame'>
RangeIndex: 2112000 entries, 0 to 2111999
Data columns (total 34 columns):
 #   Column                                      Dtype
---  ------                                      -----
 0   finalWorth                                  int64
 1   category                                    object
 2   personName                                  object
 3   age                                         int64
 4   country                                     object
 5   city                                        object
 6   source                                      object
 7   industries                                  object
 8   countryOfCitizenship                        object
 9   organization                                object
 10  selfMade                                    bool
 11  status                                      object
 12  gender                                      object
 13  birthDate                                   ob

In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["finalWorth"] = data["finalWorth"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["category"] = data["category"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["personName"] = data["personName"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["age"] = data["age"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["country"] = data["country"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["city"] = data["city"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["source"] = data["source"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["industries"] = data["industries"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["countryOfCitizenship"] = data["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["organization"] = data["organization"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["selfMade"] = data["selfMade"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["status"] = data["status"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["gender"] = data["gender"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["birthDate"] = data["birthDate"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["lastName"] = data["lastName"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["firstName"] = data["firstName"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["title"] = data["title"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["date"] = data["date"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["state"] = data["state"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["residenceStateRegion"] = data["residenceStateRegion"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["birthYear"] = data["birthYear"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["birthMonth"] = data["birthMonth"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["birthDay"] = data["birthDay"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["cpi_country"] = data["cpi_country"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["cpi_change_country"] = data["cpi_change_country"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["gdp_country"] = data["gdp_country"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["gross_tertiary_education_enrollment"] = data[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["gross_primary_education_enrollment_country"] = data[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["life_expectancy_country"] = data["life_expectancy_country"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["tax_revenue_country_country"] = data["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["total_tax_rate_country"] = data["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["population_country"] = data["population_country"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["latitude_country"] = data["latitude_country"]

In [ ]:
%%time
## Transfer_post 1 ##
pd_data["longitude_country"] = data["longitude_country"]

In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["finalWorth"] = data["finalWorth"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["category"] = data["category"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["personName"] = data["personName"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["age"] = data["age"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["country"] = data["country"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["city"] = data["city"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["source"] = data["source"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["industries"] = data["industries"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["countryOfCitizenship"] = data["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["organization"] = data["organization"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["selfMade"] = data["selfMade"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["status"] = data["status"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["gender"] = data["gender"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["birthDate"] = data["birthDate"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["lastName"] = data["lastName"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["firstName"] = data["firstName"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["title"] = data["title"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["date"] = data["date"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["state"] = data["state"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["residenceStateRegion"] = data["residenceStateRegion"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["birthYear"] = data["birthYear"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["birthMonth"] = data["birthMonth"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["birthDay"] = data["birthDay"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["cpi_country"] = data["cpi_country"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["cpi_change_country"] = data["cpi_change_country"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["gdp_country"] = data["gdp_country"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["gross_tertiary_education_enrollment"] = data[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["gross_primary_education_enrollment_country"] = data[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["life_expectancy_country"] = data["life_expectancy_country"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["tax_revenue_country_country"] = data["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["total_tax_rate_country"] = data["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["population_country"] = data["population_country"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["latitude_country"] = data["latitude_country"]

In [ ]:
%%time
## Transfer_pre 2 ##
pd_data["longitude_country"] = data["longitude_country"]

In [6]:
%%time
### cell 2 ###

data.head(5)

CPU times: user 11.3 ms, sys: 648 µs, total: 12 ms
Wall time: 9.54 ms


,finalWorth,category,personName,age,country,city,source,industries,countryOfCitizenship,organization,...,cpi_change_country,gdp_country,gross_tertiary_education_enrollment,gross_primary_education_enrollment_country,life_expectancy_country,tax_revenue_country_country,total_tax_rate_country,population_country,latitude_country,longitude_country
0,211000,Fashion & Retail,Bernard Arnault & family,74,France,Paris,LVMH,Fashion & Retail,France,LVMH Moët Hennessy Louis Vuitton,...,1.1,"$2,715,518,274,227",65.6,102.5,82.5,24.2,60.7,67059887,46.227638,2.213749
1,180000,Automotive,Elon Musk,51,United States,Austin,"Tesla, SpaceX",Automotive,United States,Tesla,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523,37.090240,-95.712891
2,114000,Technology,Jeff Bezos,59,United States,Medina,Amazon,Technology,United States,Amazon,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523,37.090240,-95.712891
3,107000,Technology,Larry Ellison,78,United States,Lanai,Oracle,Technology,United States,Oracle,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523,37.090240,-95.712891
4,106000,Finance & Investments,Warren Buffett,92,United States,Omaha,Berkshire Hathaway,Finance & Investments,United States,Berkshire Hathaway Inc. (Cl A),...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523,37.090240,-95.712891


In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["finalWorth"] = data["finalWorth"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["category"] = data["category"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["personName"] = data["personName"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["age"] = data["age"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["country"] = data["country"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["city"] = data["city"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["source"] = data["source"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["industries"] = data["industries"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["countryOfCitizenship"] = data["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["organization"] = data["organization"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["selfMade"] = data["selfMade"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["status"] = data["status"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["gender"] = data["gender"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["birthDate"] = data["birthDate"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["lastName"] = data["lastName"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["firstName"] = data["firstName"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["title"] = data["title"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["date"] = data["date"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["state"] = data["state"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["residenceStateRegion"] = data["residenceStateRegion"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["birthYear"] = data["birthYear"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["birthMonth"] = data["birthMonth"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["birthDay"] = data["birthDay"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["cpi_country"] = data["cpi_country"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["cpi_change_country"] = data["cpi_change_country"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["gdp_country"] = data["gdp_country"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["gross_tertiary_education_enrollment"] = data[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["gross_primary_education_enrollment_country"] = data[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["life_expectancy_country"] = data["life_expectancy_country"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["tax_revenue_country_country"] = data["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["total_tax_rate_country"] = data["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["population_country"] = data["population_country"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["latitude_country"] = data["latitude_country"]

In [ ]:
%%time
## Transfer_pre 3 ##
pd_data["longitude_country"] = data["longitude_country"]

In [7]:
%%time
### cell 3 ###

data.describe()

CPU times: user 701 ms, sys: 177 ms, total: 878 ms
Wall time: 805 ms


,finalWorth,age,birthYear,birthMonth,birthDay,cpi_country,cpi_change_country,gross_tertiary_education_enrollment,gross_primary_education_enrollment_country,life_expectancy_country,tax_revenue_country_country,total_tax_rate_country,population_country,latitude_country,longitude_country
count,2.112000e+06,2.060000e+06,2.051200e+06,2.051200e+06,2.051200e+06,1.964800e+06,1.964800e+06,1.966400e+06,1.967200e+06,1.966400e+06,1.965600e+06,1.966400e+06,1.980800e+06,1.980800e+06,1.980800e+06
mean,4.623788e+03,6.514019e+01,1.957183e+03,5.740250e+00,1.209984e+01,1.277552e+02,4.364169e+00,6.722567e+01,1.028585e+02,7.812282e+01,1.254623e+01,4.396334e+01,5.102053e+08,3.490359e+01,1.258316e+01
std,9.832381e+03,1.325553e+01,1.327993e+01,3.709362e+00,9.916944e+00,2.644757e+01,3.623026e+00,2.133909e+01,4.710020e+00,3.729341e+00,5.367534e+00,1.214283e+01,5.541329e+08,1.700007e+01,8.674549e+01
min,1.000000e+03,1.800000e+01,1.921000e+03,1.000000e+00,1.000000e+00,9.955000e+01,-1.900000e+00,4.000000e+00,8.470000e+01,5.430000e+01,1.000000e-01,9.900000e+00,3.801900e+04,-4.090056e+01,-1.063468e+02
25%,1.500000e+03,5.600000e+01,1.948000e+03,2.000000e+00,1.000000e+00,1.172400e+02,1.700000e+00,5.060000e+01,1.002000e+02,7.700000e+01,9.600000e+00,3.660000e+01,6.683440e+07,3.586166e+01,-9.571289e+01
50%,2.300000e+03,6.500000e+01,1.957000e+03,6.000000e+00,1.100000e+01,1.172400e+02,2.900000e+00,6.560000e+01,1.018000e+02,7.850000e+01,9.600000e+00,4.120000e+01,3.282395e+08,3.709024e+01,1.045153e+01
75%,4.200000e+03,7.500000e+01,1.966000e+03,9.000000e+00,2.100000e+01,1.250800e+02,7.500000e+00,8.820000e+01,1.026000e+02,8.090000e+01,1.280000e+01,5.910000e+01,1.366418e+09,4.046367e+01,1.041954e+02
max,2.110000e+05,1.010000e+02,2.004000e+03,1.200000e+01,3.100000e+01,2.885700e+02,5.350000e+01,1.366000e+02,1.421000e+02,8.420000e+01,3.720000e+01,1.063000e+02,1.397715e+09,6.192411e+01,1.748860e+02


In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 4 ##
pd_data["country"] = data["country"]

In [8]:
%%time
### cell 4 ###

country_names = data[
    "country"
].value_counts()  # List of how many billionaires there are in the country

CPU times: user 21 ms, sys: 0 ns, total: 21 ms
Wall time: 19 ms


In [ ]:
%%time
## Transfer_post 4 ##
pd_country_names = country_names

In [ ]:
%%time
## Transfer_pre 5 ##
pd_country_names = country_names

In [9]:
%%time
### cell 5 ###

country_names

CPU times: user 3 µs, sys: 0 ns, total: 3 µs
Wall time: 5.96 µs


country
United States               603200
China                       418400
India                       125600
Germany                      81600
United Kingdom               65600
                             ...  
Portugal                       800
Tanzania                       800
Turks and Caicos Islands       800
Uruguay                        800
Uzbekistan                     800
Name: count, Length: 78, dtype: int64

> > > > >  ****10 countries with the most billionaires****

In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 6 ##
pd_data["category"] = data["category"]

In [ ]:
%%time
## Transfer_pre 6 ##
pd_data["country"] = data["country"]

In [ ]:
%%time
## Transfer_pre 6 ##
pd_data["finalWorth"] = data["finalWorth"]

In [10]:
%%time
### cell 6 ###

data_100 = data.loc[:100, ["finalWorth", "category", "country"]]

CPU times: user 1.64 ms, sys: 329 µs, total: 1.97 ms
Wall time: 1.51 ms


In [ ]:
pd_data_100 = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_post 6 ##
pd_data_100["finalWorth"] = data_100["finalWorth"]

In [ ]:
%%time
## Transfer_post 6 ##
pd_data_100["category"] = data_100["category"]

In [ ]:
%%time
## Transfer_post 6 ##
pd_data_100["country"] = data_100["country"]

In [ ]:
pd_data_100 = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 7 ##
pd_data_100["category"] = data_100["category"]

In [11]:
%%time
### cell 7 ###

data_100_category = data_100["category"].value_counts()

CPU times: user 10.4 ms, sys: 469 µs, total: 10.9 ms
Wall time: 9.68 ms


> > > > > ****Most Richest 100 person in the world and their categories****

In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 8 ##
pd_data["country"] = data["country"]

In [12]:
%%time
### cell 8 ###

data_usa = data[
    data["country"] == "United States"
]  # We focus on billionaires based in United States

CPU times: user 30.7 ms, sys: 17.1 ms, total: 47.8 ms
Wall time: 37.8 ms


In [ ]:
pd_data_usa = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["finalWorth"] = data_usa["finalWorth"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["category"] = data_usa["category"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["personName"] = data_usa["personName"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["age"] = data_usa["age"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["country"] = data_usa["country"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["city"] = data_usa["city"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["source"] = data_usa["source"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["industries"] = data_usa["industries"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["countryOfCitizenship"] = data_usa["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["organization"] = data_usa["organization"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["selfMade"] = data_usa["selfMade"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["status"] = data_usa["status"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["gender"] = data_usa["gender"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["birthDate"] = data_usa["birthDate"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["lastName"] = data_usa["lastName"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["firstName"] = data_usa["firstName"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["title"] = data_usa["title"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["date"] = data_usa["date"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["state"] = data_usa["state"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["residenceStateRegion"] = data_usa["residenceStateRegion"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["birthYear"] = data_usa["birthYear"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["birthMonth"] = data_usa["birthMonth"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["birthDay"] = data_usa["birthDay"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["cpi_country"] = data_usa["cpi_country"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["cpi_change_country"] = data_usa["cpi_change_country"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["gdp_country"] = data_usa["gdp_country"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["gross_tertiary_education_enrollment"] = data_usa[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["gross_primary_education_enrollment_country"] = data_usa[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["life_expectancy_country"] = data_usa["life_expectancy_country"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["tax_revenue_country_country"] = data_usa["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["total_tax_rate_country"] = data_usa["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["population_country"] = data_usa["population_country"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["latitude_country"] = data_usa["latitude_country"]

In [ ]:
%%time
## Transfer_post 8 ##
pd_data_usa["longitude_country"] = data_usa["longitude_country"]

In [ ]:
pd_data_usa = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 9 ##
pd_data_usa["category"] = data_usa["category"]

In [13]:
%%time
### cell 9 ###

data_usa_category = data_usa["category"].value_counts()
data_usa_category

CPU times: user 11.1 ms, sys: 614 µs, total: 11.8 ms
Wall time: 10.1 ms


category
Finance & Investments         152000
Technology                    112800
Food & Beverage                59200
Fashion & Retail               50400
Real Estate                    40800
Media & Entertainment          29600
Energy                         28000
Sports                         28000
Healthcare                     27200
Manufacturing                  22400
Service                        15200
Diversified                    10400
Automotive                      9600
Gambling & Casinos              4800
Construction & Engineering      4000
Logistics                       4000
Telecom                         4000
Metals & Mining                  800
Name: count, dtype: int64

> > > > > ****All Billionaires in the USA and their categories****

In [ ]:
pd_data_usa = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 10 ##
pd_data_usa["city"] = data_usa["city"]

In [14]:
%%time
### cell 10 ###

data_usa_city = data_usa["city"].value_counts()
print(data_usa_city)
data_usa_city.info()

city
New York            79200
San Francisco       29600
Los Angeles         27200
Palm Beach          16800
Dallas              16000
                    ...  
Windermere            800
Wynnewood             800
Yorktown Heights      800
Youngstown            800
Ypsilanti             800
Name: count, Length: 268, dtype: int64
<class 'pandas.core.series.Series'>
Index: 268 entries, New York to Ypsilanti
Series name: count
Non-Null Count  Dtype
--------------  -----
268 non-null    int64
dtypes: int64(1)
memory usage: 4.2+ KB
CPU times: user 23.5 ms, sys: 4.7 ms, total: 28.2 ms
Wall time: 25.7 ms


268 city exist, I'll try .head(20)

> > > > > **The 20 cities has most billionaires in usa**

In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["finalWorth"] = data["finalWorth"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["category"] = data["category"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["personName"] = data["personName"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["age"] = data["age"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["country"] = data["country"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["city"] = data["city"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["source"] = data["source"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["industries"] = data["industries"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["countryOfCitizenship"] = data["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["organization"] = data["organization"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["selfMade"] = data["selfMade"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["status"] = data["status"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["gender"] = data["gender"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["birthDate"] = data["birthDate"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["lastName"] = data["lastName"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["firstName"] = data["firstName"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["title"] = data["title"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["date"] = data["date"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["state"] = data["state"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["residenceStateRegion"] = data["residenceStateRegion"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["birthYear"] = data["birthYear"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["birthMonth"] = data["birthMonth"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["birthDay"] = data["birthDay"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["cpi_country"] = data["cpi_country"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["cpi_change_country"] = data["cpi_change_country"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["gdp_country"] = data["gdp_country"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["gross_tertiary_education_enrollment"] = data[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["gross_primary_education_enrollment_country"] = data[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["life_expectancy_country"] = data["life_expectancy_country"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["tax_revenue_country_country"] = data["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["total_tax_rate_country"] = data["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["population_country"] = data["population_country"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["latitude_country"] = data["latitude_country"]

In [ ]:
%%time
## Transfer_pre 11 ##
pd_data["longitude_country"] = data["longitude_country"]

In [15]:
%%time
### cell 11 ###

data.head(5)

CPU times: user 11.3 ms, sys: 211 µs, total: 11.5 ms
Wall time: 8.56 ms


,finalWorth,category,personName,age,country,city,source,industries,countryOfCitizenship,organization,...,cpi_change_country,gdp_country,gross_tertiary_education_enrollment,gross_primary_education_enrollment_country,life_expectancy_country,tax_revenue_country_country,total_tax_rate_country,population_country,latitude_country,longitude_country
0,211000,Fashion & Retail,Bernard Arnault & family,74,France,Paris,LVMH,Fashion & Retail,France,LVMH Moët Hennessy Louis Vuitton,...,1.1,"$2,715,518,274,227",65.6,102.5,82.5,24.2,60.7,67059887,46.227638,2.213749
1,180000,Automotive,Elon Musk,51,United States,Austin,"Tesla, SpaceX",Automotive,United States,Tesla,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523,37.090240,-95.712891
2,114000,Technology,Jeff Bezos,59,United States,Medina,Amazon,Technology,United States,Amazon,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523,37.090240,-95.712891
3,107000,Technology,Larry Ellison,78,United States,Lanai,Oracle,Technology,United States,Oracle,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523,37.090240,-95.712891
4,106000,Finance & Investments,Warren Buffett,92,United States,Omaha,Berkshire Hathaway,Finance & Investments,United States,Berkshire Hathaway Inc. (Cl A),...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523,37.090240,-95.712891


In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 12 ##
pd_data["category"] = data["category"]

In [16]:
%%time
### cell 12 ###

category_list = list(data.category.unique())
category_list

CPU times: user 159 ms, sys: 48.1 ms, total: 208 ms
Wall time: 206 ms


['Fashion & Retail',
 'Automotive',
 'Technology',
 'Finance & Investments',
 'Media & Entertainment',
 'Telecom',
 'Diversified',
 'Food & Beverage',
 'Logistics',
 'Gambling & Casinos',
 'Manufacturing',
 'Real Estate',
 'Metals & Mining',
 'Energy',
 'Healthcare',
 'Service',
 'Construction & Engineering',
 'Sports']

In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 13 ##
pd_data["category"] = data["category"]

In [ ]:
%%time
## Transfer_pre 13 ##
pd_data["finalWorth"] = data["finalWorth"]

In [17]:
%%time
### cell 13 ###

data.finalWorth = data.finalWorth.astype(float)
worth_average = []
for i in category_list:
    x = data[data["category"] == i]
    worth_ratio = sum(x.finalWorth) / len(x)
    worth_average.append(worth_ratio)
data2 = pd.DataFrame({"category_list": category_list, "worth_average": worth_average})
new_index = (data2["worth_average"].sort_values(ascending=False)).index.values
new_data = data2.reindex(new_index)

CPU times: user 1.37 s, sys: 146 ms, total: 1.51 s
Wall time: 1.44 s


In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_post 13 ##
pd_data["finalWorth"] = data["finalWorth"]

In [ ]:
pd_data = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["finalWorth"] = data["finalWorth"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["category"] = data["category"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["personName"] = data["personName"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["age"] = data["age"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["country"] = data["country"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["city"] = data["city"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["source"] = data["source"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["industries"] = data["industries"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["countryOfCitizenship"] = data["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["organization"] = data["organization"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["selfMade"] = data["selfMade"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["status"] = data["status"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["gender"] = data["gender"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["birthDate"] = data["birthDate"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["lastName"] = data["lastName"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["firstName"] = data["firstName"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["title"] = data["title"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["date"] = data["date"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["state"] = data["state"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["residenceStateRegion"] = data["residenceStateRegion"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["birthYear"] = data["birthYear"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["birthMonth"] = data["birthMonth"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["birthDay"] = data["birthDay"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["cpi_country"] = data["cpi_country"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["cpi_change_country"] = data["cpi_change_country"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["gdp_country"] = data["gdp_country"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["gross_tertiary_education_enrollment"] = data[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["gross_primary_education_enrollment_country"] = data[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["life_expectancy_country"] = data["life_expectancy_country"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["tax_revenue_country_country"] = data["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["total_tax_rate_country"] = data["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["population_country"] = data["population_country"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["latitude_country"] = data["latitude_country"]

In [ ]:
%%time
## Transfer_pre 14 ##
pd_data["longitude_country"] = data["longitude_country"]

In [18]:
%%time
### cell 14 ###

# data3 =data.dropna(axis="index",how="any",inplace=False) # ı created new df because so many nan values was exist
# for cudf doesnt return the same df
data3 = data.dropna(axis=0, how="any", subset=data.columns)

CPU times: user 3.53 s, sys: 1.42 s, total: 4.95 s
Wall time: 4.95 s


In [ ]:
pd_data3 = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["finalWorth"] = data3["finalWorth"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["category"] = data3["category"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["personName"] = data3["personName"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["age"] = data3["age"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["country"] = data3["country"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["city"] = data3["city"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["source"] = data3["source"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["industries"] = data3["industries"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["countryOfCitizenship"] = data3["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["organization"] = data3["organization"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["selfMade"] = data3["selfMade"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["status"] = data3["status"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["gender"] = data3["gender"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["birthDate"] = data3["birthDate"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["lastName"] = data3["lastName"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["firstName"] = data3["firstName"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["title"] = data3["title"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["date"] = data3["date"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["state"] = data3["state"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["residenceStateRegion"] = data3["residenceStateRegion"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["birthYear"] = data3["birthYear"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["birthMonth"] = data3["birthMonth"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["birthDay"] = data3["birthDay"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["cpi_country"] = data3["cpi_country"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["cpi_change_country"] = data3["cpi_change_country"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["gdp_country"] = data3["gdp_country"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["gross_tertiary_education_enrollment"] = data3[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["gross_primary_education_enrollment_country"] = data3[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["life_expectancy_country"] = data3["life_expectancy_country"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["tax_revenue_country_country"] = data3["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["total_tax_rate_country"] = data3["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["population_country"] = data3["population_country"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["latitude_country"] = data3["latitude_country"]

In [ ]:
%%time
## Transfer_post 14 ##
pd_data3["longitude_country"] = data3["longitude_country"]

In [ ]:
pd_data3 = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["finalWorth"] = data3["finalWorth"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["category"] = data3["category"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["personName"] = data3["personName"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["age"] = data3["age"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["country"] = data3["country"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["city"] = data3["city"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["source"] = data3["source"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["industries"] = data3["industries"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["countryOfCitizenship"] = data3["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["organization"] = data3["organization"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["selfMade"] = data3["selfMade"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["status"] = data3["status"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["gender"] = data3["gender"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["birthDate"] = data3["birthDate"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["lastName"] = data3["lastName"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["firstName"] = data3["firstName"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["title"] = data3["title"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["date"] = data3["date"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["state"] = data3["state"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["residenceStateRegion"] = data3["residenceStateRegion"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["birthYear"] = data3["birthYear"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["birthMonth"] = data3["birthMonth"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["birthDay"] = data3["birthDay"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["cpi_country"] = data3["cpi_country"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["cpi_change_country"] = data3["cpi_change_country"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["gdp_country"] = data3["gdp_country"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["gross_tertiary_education_enrollment"] = data3[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["gross_primary_education_enrollment_country"] = data3[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["life_expectancy_country"] = data3["life_expectancy_country"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["tax_revenue_country_country"] = data3["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["total_tax_rate_country"] = data3["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["population_country"] = data3["population_country"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["latitude_country"] = data3["latitude_country"]

In [ ]:
%%time
## Transfer_pre 15 ##
pd_data3["longitude_country"] = data3["longitude_country"]

In [19]:
%%time
### cell 15 ###

data3.head()

CPU times: user 1.03 s, sys: 8.59 ms, total: 1.04 s
Wall time: 1.03 s


,finalWorth,category,personName,age,country,city,source,industries,countryOfCitizenship,organization,...,cpi_change_country,gdp_country,gross_tertiary_education_enrollment,gross_primary_education_enrollment_country,life_expectancy_country,tax_revenue_country_country,total_tax_rate_country,population_country,latitude_country,longitude_country
1,180000.0,Automotive,Elon Musk,51.0,United States,Austin,"Tesla, SpaceX",Automotive,United States,Tesla,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523.0,37.09024,-95.712891
2,114000.0,Technology,Jeff Bezos,59.0,United States,Medina,Amazon,Technology,United States,Amazon,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523.0,37.09024,-95.712891
3,107000.0,Technology,Larry Ellison,78.0,United States,Lanai,Oracle,Technology,United States,Oracle,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523.0,37.09024,-95.712891
4,106000.0,Finance & Investments,Warren Buffett,92.0,United States,Omaha,Berkshire Hathaway,Finance & Investments,United States,Berkshire Hathaway Inc. (Cl A),...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523.0,37.09024,-95.712891
5,104000.0,Technology,Bill Gates,67.0,United States,Medina,Microsoft,Technology,United States,Bill & Melinda Gates Foundation,...,7.5,"$21,427,700,000,000",88.2,101.8,78.5,9.6,36.6,328239523.0,37.09024,-95.712891


In [ ]:
pd_data3 = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["finalWorth"] = data3["finalWorth"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["category"] = data3["category"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["personName"] = data3["personName"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["age"] = data3["age"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["country"] = data3["country"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["city"] = data3["city"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["source"] = data3["source"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["industries"] = data3["industries"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["countryOfCitizenship"] = data3["countryOfCitizenship"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["organization"] = data3["organization"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["selfMade"] = data3["selfMade"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["status"] = data3["status"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["gender"] = data3["gender"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["birthDate"] = data3["birthDate"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["lastName"] = data3["lastName"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["firstName"] = data3["firstName"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["title"] = data3["title"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["date"] = data3["date"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["state"] = data3["state"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["residenceStateRegion"] = data3["residenceStateRegion"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["birthYear"] = data3["birthYear"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["birthMonth"] = data3["birthMonth"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["birthDay"] = data3["birthDay"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["cpi_country"] = data3["cpi_country"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["cpi_change_country"] = data3["cpi_change_country"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["gdp_country"] = data3["gdp_country"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["gross_tertiary_education_enrollment"] = data3[
    "gross_tertiary_education_enrollment"
]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["gross_primary_education_enrollment_country"] = data3[
    "gross_primary_education_enrollment_country"
]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["life_expectancy_country"] = data3["life_expectancy_country"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["tax_revenue_country_country"] = data3["tax_revenue_country_country"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["total_tax_rate_country"] = data3["total_tax_rate_country"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["population_country"] = data3["population_country"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["latitude_country"] = data3["latitude_country"]

In [ ]:
%%time
## Transfer_pre 16 ##
pd_data3["longitude_country"] = data3["longitude_country"]

In [20]:
%%time
### cell 16 ###

data3.info()  # in the new df we have 238 rows for each column

<class 'cudf.core.dataframe.DataFrame'>
Index: 190400 entries, 1 to 2111968
Data columns (total 34 columns):
 #   Column                                      Non-Null Count   Dtype
---  ------                                      --------------   -----
 0   finalWorth                                  190400 non-null  float64
 1   category                                    190400 non-null  object
 2   personName                                  190400 non-null  object
 3   age                                         190400 non-null  float64
 4   country                                     190400 non-null  object
 5   city                                        190400 non-null  object
 6   source                                      190400 non-null  object
 7   industries                                  190400 non-null  object
 8   countryOfCitizenship                        190400 non-null  object
 9   organization                                190400 non-null  object
 10  selfMade  

In [ ]:
pd_data3 = cudf.DataFrame().to_pandas()

In [ ]:
%%time
## Transfer_pre 17 ##
pd_data3["countryOfCitizenship"] = data3["countryOfCitizenship"]

In [21]:
%%time
### cell 17 ###

data3.countryOfCitizenship.unique()  # and 238 billionaires in usa, others deleted.

CPU times: user 18.6 ms, sys: 0 ns, total: 18.6 ms
Wall time: 17.7 ms


array(['United States', 'Canada', 'Germany', 'Israel', 'Norway',
       'Ireland', 'Turkey', 'France'], dtype=object)